pallets/nodle-staking/src/migrations.rs

```diff
@@ -0,0 +1,297 @@
+/*
+ * This file is part of the Nodle Chain distributed at https://github.com/NodleCode/chain
+ * Copyright (C) 2020  Nodle International
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+use super::{BalanceOf, Config, Validator};
+
+use frame_support::{
+    pallet_prelude::*,
+    traits::StorageInstance,
+    weights::{constants::RocksDbWeight, Weight},
+};
+
+use sp_runtime::traits::Zero;
+use sp_std::vec::Vec;
+
+/// This migration is in charge of moving the validators from POA to Staking pallet to
+/// work with the last chain reboot.
+/// We have to do two things:
+/// - read "Validators" storage instance from POA.
+/// - with read info update the "Invulnerables" & "ValidatorState" of staking pallet.
+#[allow(dead_code)]
+pub fn poa_validators_migration<T: Config>() -> Weight {
+    // Buffer to account for misc checks
+    let mut weight: Weight = 1_000;
+
+    log::info!("🕊️ Starting Poa validators migration...");
+
+    struct POAValidatorsStorageValuePrefix;
+    impl StorageInstance for POAValidatorsStorageValuePrefix {
+        const STORAGE_PREFIX: &'static str = "Validators";
+        fn pallet_prefix() -> &'static str {
+            "Poa"
+        }
+    }
+
+    struct StakingValidatorStateStorageMapPrefix;
+    impl StorageInstance for StakingValidatorStateStorageMapPrefix {
+        const STORAGE_PREFIX: &'static str = "ValidatorState";
+        fn pallet_prefix() -> &'static str {
+            "NodleStaking"
+        }
+    }
+
+    struct StakingInvulnerablesStorageValuePrefix;
+    impl StorageInstance for StakingInvulnerablesStorageValuePrefix {
+        const STORAGE_PREFIX: &'static str = "Invulnerables";
+        fn pallet_prefix() -> &'static str {
+            "NodleStaking"
+        }
+    }
```

Comment:
Do we absolutely need to duplicate this?

Reply:
HaHa, No Eliott they are not duplicate.  

* First one is referring to storage instance

```
pub(crate) type Invulnerables<T: Config> = StorageValue<_, Vec<T::AccountId>, ValueQuery>;
```

* Second one is referring to storage instance

```
pub(crate) type ValidatorState<T: Config> = StorageMap<>
```